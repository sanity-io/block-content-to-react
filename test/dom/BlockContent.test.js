/* eslint-disable id-length, max-len */
const React = require('react')
const ReactDOM = require('react-dom/server')
const runTests = require('@sanity/block-content-tests')
const BlockContent = require('../../src/BlockContent')

const h = React.createElement
const getImageUrl = BlockContent.getImageUrl
const render = props => ReactDOM.renderToStaticMarkup(h(BlockContent, props))
const normalize = html =>
  html.replace(/ style="(.*?)"/g, (match, styleProps) => {
    const style = styleProps.replace(/;$/g, '')
    return ` style="${style}"`
  })

runTests({render, h, normalize, getImageUrl})

test('renders empty block with react proptype error', () => {
  expect(render({})).toMatchSnapshot()
})
