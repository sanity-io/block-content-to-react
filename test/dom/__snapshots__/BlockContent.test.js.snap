// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty block with react proptype error 1`] = `""`;
